sql/tests: TestRandomSyntaxSQLSmith failed

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/13152786?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/13152786?buildTab=artifacts#/) on release-22.2 @ [8f36d6b43206f1d553bb0b2ffc50702ca69767c1](https://github.com/cockroachdb/cockroach/commits/8f36d6b43206f1d553bb0b2ffc50702ca69767c1):

Random syntax error:

```
    rsg_test.go:854: Crash detected: server panic: statement exec timeout
```
Query:

```
        SELECT
        	(-13503):::INT8 AS col_521960,
        	crdb_internal.reset_sql_stats()::BOOL AS col_521961,
        	tab_221118._decimal AS col_521962,
        	NULL AS col_521963,
        	tab_221119._int2 AS col_521964,
        	7727486533679031.582:::DECIMAL AS col_521965,
        	jsonb_insert(tab_221115._jsonb::JSONB, NULL::STRING[], e'{"0``m+J3BxNI": [2.500942627451038, [[[], {}], {"foobar": "2v@"}, {}, 0.463644556177129]], "a": {"6Tm_<SY": null, "8b+Azec2Yy": {"^xM\'}4gsyRP": {}}}, "|Zs]?V)Sq": []}':::JSONB::JSONB, tab_221118._bool::BOOL)::JSONB
        		AS col_521970,
        	tab_221118._float4 AS col_521971,
        	tab_221118.tableoid AS col_521972,
        	NULL AS col_521973,
        	'680e28a0-e083-45f8-b502-fa0cb3ace8eb':::UUID AS col_521974,
        	tab_221118._interval AS col_521975,
        	'17c5c899-21d9-477d-b87d-e1c8452169ea':::UUID AS col_521976,
        	NULL AS col_521977,
        	COALESCE((NULL::JSONB->e'yf\x7fo\x044\rh':::STRING::STRING)::JSONB, tab_221116._jsonb) AS col_521978,
        	true AS col_521979,
        	'1970-03-04 02:21:08.000616+00:00':::TIMESTAMPTZ AS col_521980,
        	(-80196260.6424304845):::DECIMAL AS col_521981,
        	tab_221119._bool AS col_521982,
        	'1991-08-30 22:31:17.000118+00:00':::TIMESTAMPTZ AS col_521983,
        	tab_221119._jsonb AS col_521984,
        	tab_221115._int4 AS col_521985
        FROM
        	defaultdb.public.seed@seed__int8__float8__date_idx AS tab_221115,
        	defaultdb.public.seed AS tab_221116
        	JOIN defaultdb.public.seed AS tab_221117 ON (tab_221116._jsonb) = (tab_221117._jsonb),
        	defaultdb.public.seed@seed__int8__float8__date_idx AS tab_221118,
        	defaultdb.public.seed@[0] AS tab_221119
        ORDER BY
        	tab_221119._decimal DESC NULLS FIRST
        LIMIT
        	33:::INT8;
```
Schema:

```
    rsg_test.go:668: To reproduce, use schema:
    rsg_test.go:670: SET sql_safe_updates = false;;
    rsg_test.go:670: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:670: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:670: CREATE TABLE table1 (col1_0 BIT(2) NOT NULL, col1_1 TIME NOT NULL, col1_2 REGCLASS NULL, col1_3 TIME, col1_4 TIMETZ NOT NULL, col1_5 REGTYPE NOT NULL, col1_6 REGCLASS NULL, col1_7 DECIMAL NOT NULL, col1_8 OID, col1_9 STRING COLLATE en_US NOT NULL, col1_10 FLOAT8 NOT NULL, col1_11 JSONB NOT NULL, col1_12 INT8 NULL, col1_13 BYTES NOT NULL, col1_14 UUID[], col1_15 INT2, col1_16 NAME NOT NULL, col1_17 INT8 NULL AS (col1_12 + col1_15) VIRTUAL, col1_18 INT8 NULL AS (col1_12 + col1_15) STORED, PRIMARY KEY (col1_9 ASC, col1_5, col1_1 DESC, col1_0 DESC, col1_13 DESC, col1_4), FAMILY (col1_9), FAMILY (col1_2), FAMILY (col1_4), FAMILY (col1_8), FAMILY (col1_5, col1_1, col1_14), FAMILY (col1_13, col1_11, col1_10), FAMILY (col1_0, col1_7), FAMILY (col1_3, col1_15, col1_12, col1_16), FAMILY (col1_6), FAMILY (col1_18));
    rsg_test.go:670: ALTER TABLE table1 INJECT STATISTICS '[{"avg_size": 17, "columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6, "histo_col_type": "", "name": "__auto__", "null_count": 3, "row_count": 30}, {"avg_size": 12, "columns": ["col1_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 25, "histo_col_type": "", "name": "__auto__", "null_count": 23, "row_count": 30}, {"avg_size": 17, "columns": ["col1_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4, "histo_col_type": "", "name": "__auto__", "null_count": 25, "row_count": 30}, {"avg_size": 23, "columns": ["col1_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 18, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 30}, {"avg_size": 2, "columns": ["col1_12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 19, "histo_col_type": "", "name": "__auto__", "null_count": 21, "row_count": 30}, {"avg_size": 13, "columns": ["col1_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 21, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 30}, {"avg_size": 12, "columns": ["col1_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 11, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 30}, {"avg_size": 23, "columns": ["col1_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 12, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 30}, {"avg_size": 8, "columns": ["col1_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 30}, {"avg_size": 1, "columns": ["col1_17"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 13, "histo_col_type": "", "name": "__auto__", "null_count": 17, "row_count": 30}, {"avg_size": 14, "columns": ["col1_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 26, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 30}, {"avg_size": 19, "columns": ["col1_14"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 11, "histo_col_type": "", "name": "__auto__", "null_count": 6, "row_count": 30}, {"avg_size": 2, "columns": ["col1_16"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 20, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 30}, {"avg_size": 0, "columns": ["col1_18"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 26, "histo_col_type": "", "name": "__auto__", "null_count": 29, "row_count": 30}, {"avg_size": 8, "columns": ["col1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 11, "histo_col_type": "", "name": "__auto__", "null_count": 21, "row_count": 30}, {"avg_size": 9, "columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 30}, {"avg_size": 4, "columns": ["col1_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 30}, {"avg_size": 9, "columns": ["col1_13"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 18, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 30}, {"avg_size": 30, "columns": ["col1_15"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 7, "histo_col_type": "", "name": "__auto__", "null_count": 10, "row_count": 30}]':::JSONB;
    rsg_test.go:670: CREATE TYPE rand_typ_0 AS ENUM ('i', 'd', 'ce', 'm', 'auq');
    rsg_test.go:670: SET sql_safe_updates = false;;
    rsg_test.go:670: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:672: 
    rsg_test.go:673: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith3423864569
--- FAIL: TestRandomSyntaxSQLSmith (301.57s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34653

--- a/pkg/internal/sqlsmith/schema.go
+++ b/pkg/internal/sqlsmith/schema.go
@@ -525,6 +525,7 @@ var functions = func() *functionsMu {
 			"crdb_internal.revalidate_unique_constraint",
 			"crdb_internal.request_statement_bundle",
 			"crdb_internal.set_compaction_concurrency",
+			"crdb_internal.reset_sql_stats",
 		} {
 			skip = skip || strings.Contains(def.Name, substr)
 		}

